Update benchmark readme with info on how to ensure benchmarks test locally built stdlib (#79528)
swiftlang/swift: benchmark/README.md

This will reflect the performance of the Swift standard library
installed on the device, not the one included in the Swift root.

> If build fails due with the error like `cannot find X in scope` it my be because default toolchains don't give access to stdlib SPI.
> Try using dev toolchain built locally or dev snapshot from [swift.org](https://www.swift.org/install/).
> 

Reviewer: ```suggestion
If a build fails with an error like `cannot find X in scope`, it may be because default toolchains don't give access to stdlib SPI.
Try using a dev toolchain built locally or a dev snapshot from [Swift.org](https://www.swift.org/install/).
```

Reviewer: Would this not also better fit into the troubleshooting section?

Author: I don't have a strong preference.
Current troubleshooting section is global for all builds. I had to apply those fixes for benchmarks that are built as part of overall build.
This particular advice about toolchain is only applicable to the standalone built via Cmake. This is why I put it here.
LMK if you want me to move it.

> built locally or a dev snapshot from [Swift.org](https://www.swift.org/install/).

Reviewer: This line is still longer.

Author: I thought it would mess up rendering. But I was wrong. Will update.

...
```

### Troubleshooting

> - To check what libraries benchmark is linked against set `DYLD_PRINT_SEARCHING` environment variable. E.g. `export DYLD_PRINT_SEARCHING=1`.
> - To force linking with built stdlib use `DYLD_LIBRARY_PATH` environment variable. E.g. `export DYLD_LIBRARY_PATH=<path_to_build_folder>/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx/x86_64`.

Reviewer: ```suggestion
- To check what libraries benchmark is linked against, set the `DYLD_PRINT_SEARCHING` environment variable. E.g. `export DYLD_PRINT_SEARCHING=1`.
- To force linking with a locally built stdlib, use the `DYLD_LIBRARY_PATH` environment variable. E.g. `export DYLD_LIBRARY_PATH=<path_to_build_folder>/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/macosx/x86_64`.
```
